app/helpers/cache.rb

```diff
@@ -1,15 +1,33 @@
 module Cache
-  def self.save_to_cache(key_and_values, expiration)
+  CACHE_EXPIRATION = 30.days
+
+  def self.write(key_and_values, expiration=nil)
     key_and_values.each do |k,v|
       Rails.cache.delete k
-      Rails.cache.write(k, v, expires_in: expiration)
+      Rails.cache.write(k, v, expires_in: expiration || CACHE_EXPIRATION)
     end
   end
-  def self.retrieve(key)
-    Rails.cache.read(key)
+
+  def self.get_stations(lat, lon)
+    return nil if changed_station_search?(lat, lon)
+    retrieve(:ev_stations)
+  end
+
+  def self.get_businesses(activities, lat, lon)
+    return nil if changed_business_search?(activities, lat, lon)
+    retrieve(:activities)
+  end
+
+  def self.retrieve(key, options=nil)
+    value = Rails.cache.read(key)
+    Rails.cache.delete key if options && options[:delete]
+    value
   end
+
+  private
+
   def self.changed_business_search?(activities, lat, lon)
-    activity_search = activities == Rails.cache.read('activities')
+    activity_search = activities == Rails.cache.read(:activities)
```

Comment:
this is a really great refactor of the cache, pulling all this out of the controllers cleans it up nicely.